services/resources-getter.js

```diff
@@ -150,14 +144,44 @@ function ResourcesGetter(model, opts, params) {
           var fieldsSearched = searchBuilder.getFieldsSearched();
           if (fieldsSearched.length === 0 && !hasSmartFieldSearch) {
             // NOTICE: No search condition has been set for the current search, no record can be found.
-            return [0, []];
+            return [];
           }
         }
 
-        return P.all([
-          scope.count(countOpts),
-          scope.findAll(findAllOpts)
-        ]);
+        return scope.findAll(findAllOpts);
+    });
+  }
+
+  function countRecords() {
+    var scope = segmentScope ? model.scope(segmentScope) : model.unscoped();
+    var include = queryBuilder.getIncludes(model, fieldNamesRequested);
+
+    return getWhere()
+      .then(function (where) {
+        var countOpts = {
+          include: include,
+          where: where
+        };
+
+        if (!fieldNamesRequested[0]) {
```

Comment:
This criteria is error prone IMO: `fieldNamesRequested` is not always available information (in Smart Views for example). Can't you check that the schema as a primary key? I'll try this.